### PR TITLE
docs: table sharding in user guide

### PR DESCRIPTION
**Pull request by waynexia**
## What's Changed in this PR

Closes https://github.com/GreptimeTeam/docs/issues/1046

Describe the when and how to shard a table

## Checklist

- [x] Please confirm that all corresponding versions of the documents have been revised.
- [x] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [x] This change requires follow-up update in localized docs.

## Summary by CodeRabbit

- **Documentation**
  - Added new documentation on table sharding in GreptimeDB, explaining its importance in distributed databases, guidelines for sharding tables, and partition rules based on column value ranges.
  - Updated contributor guide to focus on data sharding concepts, removing specific mentions of partitioning types.
  - Introduced a new section titled `table-sharding` in the summary of topics.

### CODE REVIEW

**Comment by waynexia** [comment]
@coderabbitai pause

**Comment by waynexia** [comment]
@coderabbitai resolve

**Review by zhaoyingnan01 — APPROVED** [review]

File: docs/nightly/en/user-guide/operations/table-sharding.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+![table-sharding-partition](/table-sharding-partition.png)
+
+Two things need stress in this complex example:
+- Each column in the partition rule are evaluated **separately**, this is different from the traditional storage system like MySQL or TiKV. Hence you can write whatever complex rule you want without considering the "primary key order" or "physical storage order".
+- String comparison is based on the lexicographical order. E.g., string `"10"` is less than string `"2"`.
+
+## Inspect a sharded table
+
+GreptimeDB provides severals system table to check DB's state. For table sharding information, you can query [`information_schema.partitions`](../../reference/sql/information-schema/partitions.md) which gives the detail of partitions inside one table, and [`information_schema.region_peers`](../../reference/sql/information-schema/region-peers.md) which gives the runtime distribution of regions.
```
Comment (nicecui) [inline docs/nightly/en/user-guide/operations/table-sharding.md:80]:
```suggestion
GreptimeDB provides severals system table to check DB's state. For table sharding information, you can query [`information_schema.partitions`](/reference/sql/information-schema/partitions.md) which gives the detail of partitions inside one table, and [`information_schema.region_peers`](/reference/sql/information-schema/region-peers.md) which gives the runtime distribution of regions.
```

**Review by nicecui — COMMENTED** [review]
LGTM, and when @killme2008 or @fengjiachun is free, it is better to confirm all the description is correct.

File: docs/nightly/en/summary.yml [hunk 2]
```diff
@@ -76,6 +76,7 @@
     - admin
     - configuration
     - capacity-plan
+    - table-sharding
```
Comment (killme2008) [inline docs/nightly/en/summary.yml:79]: I think it's not a part of the operation but table management.

File: docs/nightly/en/user-guide/operations/table-sharding.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+A region in GreptimeDB has a relative fixed throughput capacity, and the number of regions in a table determines the total throughput capacity of the table. If you want to increase the throughput capacity of a table, you can increase the number of regions in the table. Ideally the overall throughput of a table should be proportional to the number of regions.
+
+As for which specific partition column to use or how many regions to create, it depends on the data distribution and the query pattern. A general goal is to make the data distribution among regions as even as possible. And the query pattern should be considered when designing the partition rule set as one query can be processed in parallel among regions. In other word the query latency is depends on the "slowest" region's latency.
+
+But notice that the increase of regions will bring some basic consumption and increase the complexity of the system. You need to consider the requirement of data ingest rate, the query performance, the data distribution on storage system. You should shard a table only when necessary.
+
```
Comment (killme2008) [inline docs/nightly/en/user-guide/operations/table-sharding.md:16]: Include a diagram illustrating how tables, regions, frontends, and datanodes collaborate.

File: docs/nightly/en/user-guide/operations/table-sharding.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+```sql
+CREATE TABLE (...)
+PARTITION ON COLUMNS (<COLUMN LIST>) (
+    <RULE LIST>
+);
+```
+
+The syntax mainly consists of two parts:
+- `PARTITION ON COLUMNS` followed by a comma-separated list of column names, which specifies which columns might be used for partitioning. The partition list specified here is only used as an "allow list", and in reality only a portion of the columns specified here will be used for partitioning.
```
Comment (killme2008) [inline docs/nightly/en/user-guide/operations/table-sharding.md:29]: Add a note? The partition columns should be in primary key constraint.

File: docs/nightly/en/user-guide/operations/table-sharding.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+You can also define the rule set based on multiple primary key columns. For example, if you have a table with two primary key columns `series` and `host`. And we know how the load is distributed in the table. Like:
+
+![table-sharding-load](/table-sharding-load.png)
+
+And we want to partition the table for some reason like one region's capacity is 30 load unit. So we'll need 6 partitions with each has similar load. One possible partition rule set is:
+
+```sql
+CREATE TABLE my_table (
```
Comment (killme2008) [inline docs/nightly/en/user-guide/operations/table-sharding.md:54]: Better to change the table name.

File: docs/nightly/en/user-guide/operations/table-sharding.md (hunk 1 above)
Comment (killme2008) [inline docs/nightly/en/user-guide/operations/table-sharding.md:80]: check regions state?

**Comment by waynexia** [comment]
hold this for repartition feature

**Comment by nicecui** [comment]
may be we can write a new PR for the repartition feature

**Pull request closed without merging**